### PR TITLE
Updates for Gun Mettle and other things

### PR DESCRIPTION
**Pull request by Starblaster64**
Not entirely sure how you might've wanted to handle the cloak resistances, but I went with something that should work.

### CODE REVIEW

**Comment by Starblaster64** [comment]
In retrospect, it may have been wiser to just wait and see how the new cloak stuff works first before making any changes, guess we'll have to see.

File: addons/sourcemod/scripting/saxtonhale.sp [hunk 1]
```diff
@@ -4916,20 +4916,20 @@ public Action OnTakeDamage(int client, int &attacker, int &inflictor, float &dam
             TF2_AddCondition(client, TFCond_SpeedBuffAlly, 1.0);
             return Plugin_Continue;
         }
-        if (TF2_GetPlayerClass(client) == TFClass_Spy)  //eggs probably do melee damage to spies, then? That's not ideal, but eh.
+            if ((TF2_GetPlayerClass(client) == TFClass_Spy) && (damagetype & DMG_CLUB)) //Only Melee hits get altered damage
```
Comment (WildCard65) [inline addons/sourcemod/scripting/saxtonhale.sp]: Extra tab not needed.

**Comment by Starblaster64** [comment]
Gah, just realised I'm a complete dunce. Went about the cloak thing in a stupid way. I'll fix it later, need sleep now.

**Comment by Starblaster64** [comment]
Looks like Valve went and removed the hidden Shortstop reload penalty themselves, going to keep it as it is in-case that was unintended. Also, how do we want to handle the weapon drops? So far I've found a temporary solution of setting 'tf_dropped_weapon_lifetime' to 0.

**Comment by Starblaster64** [comment]
Apparently the new weapon skins are done like Festives and not like Australiums, joy. Would rounding up all the item IDs be the method of choice, or would a strncmp method work?

**Comment by WildCard65** [comment]
Mind resolving merge conflict.

**Comment by Starblaster64** [comment]
I would, but uh, I'm not sure how. I'm pretty sure I fixed it in the "Fix'd" commit, but my knowledge of GitHub/Travis is a bit limited.

**Comment by Starblaster64** [comment]
I won't be to work on this for the next day and a bit, flying overseas and whatnot. If someone wants to just take all these changes (plus the sniper rifle thing) and make a new pull request before then I'll close this one.

**Comment by WildCard65** [comment]
hey @shadow93 want to work on this? @Starblaster64 to fix merge conflicts, you need to merge the original develop branch into this branch then manually fix the conflicts that arrise from merge then commits changes.

**Comment by Chdata** [comment]
I plan to also remove the speed buff on cloak from all watches.

**Comment by Starblaster64** [comment]
@WildCard65 ah, makes sense. I'll do that as soon as I get to my destination. Also @Chdata, has your stance on not changing the Enforcer changed after the latest update?

**Comment by Chdata** [comment]
I reverted it to the previous version in my fork, more or less. It gets 1.2x damage but 0.8x damage while disguised. Essentially if you have 0.96 damage while disguised if you stack that together.

**Comment by Starblaster64** [comment]
Going to add a fix to 'addthecrit' for affected weapons next. Anything else that needs to be updated after that?

**Comment by WildCard65** [comment]
@Starblaster64 if your using the git gui client, to easiy reverse a commit is to select said commit and click the Revert button.

**Comment by Starblaster64** [comment]
@WildCard65 I'm using SourceTree, it has a similar function but I was tired and managed to screw it up somehow, so I did it manually.

**Comment by Starblaster64** [comment]
@WildCard65 Ok, so I'm working on a better fix for the cloak stuff that involves changing some CVARs. Now either I'm doing something wrong, or the way you've done it for the other CVARs is wrong, because none of them are being changed, even the ones that I didn't add into the list. I'm using the method that the Master branch has for now, just for testing purposes.

**Comment by WildCard65** [comment]
Mind telling me what cvars your talking about?

**Comment by Starblaster64** [comment]
Derp, that was stupid

**Comment by Starblaster64** [comment]
Just tested your latest commit @WildCard65, methodmaps still aren't working. I'm going to just push my changes without using methodmaps and we can sort it out later.

**Comment by FlaminSarge** [comment]
I have no idea what's going on, but if you guys need help, let me know.

**Comment by Starblaster64** [comment]
@FlaminSarge Thanks, I'll keep that in mind next time. Still a little new to Git and GitHub.

**Comment by WildCard65** [comment]
@Starblaster64 I kind of forgot to put in change hooks for the cvars VSH modify.

**Comment by Starblaster64** [comment]
@WildCard65 I'll hold off my commit for now then.

**Comment by Chdata** [comment]
here's the list of cvars starblaster found
[SM] Value of cvar "tf_feign_death_activate_damage_scale": "0.5"
[SM] Value of cvar "tf_feign_death_damage_scale": "0.35"
[SM] Value of cvar "tf_feign_death_duration": "3.0"
[SM] Value of cvar "tf_feign_death_speed_duration": "3.0"
[SM] Value of cvar "tf_stealth_damage_reduction": "0.8"

0.9
0.9
7.0
0.0
0.9
probably would fix DR without even changing the 620/850 code.

**Comment by Starblaster64** [comment]
@Chdata I was going to do the whole thing like that, but CBS arrows wouldn't deal the constant damage that they do now.

**Comment by Chdata** [comment]
oh yeah, but that's still a small edit (Special == VSHSpecial_CBS && !(IDmgType & DMG_CLUB)) <- boss is cbs, but damage isn't melee

**Comment by Starblaster64** [comment]
That still leaves the damage scaling over time for the Dead Ringer buff.

**Comment by Chdata** [comment]
shouldn't the cvars I just mentioned make it scale from 0.9 to ... 0.9 instead of 0.65 to 0.8

**Comment by Starblaster64** [comment]
Yes, but then both the invis watch and dead ringer will have the same resist. Also I think you mean 0.1, not 0.9.

**Comment by Chdata** [comment]
yes 0.1

and the normal watch is supposed to have the same resist, because we set the damage to 850.0

**Comment by Starblaster64** [comment]
Ohhh, you mean leave the OTD damage unchanged. That makes more sense now.

**Comment by Starblaster64** [comment]
Ok, so I'm going to quickly draft out here how I'm going to do the cloak stuff, since I have to re-do my re-do now.

"tf_feign_death_activate_damage_scale": "0.9"
"tf_feign_death_damage_scale": "0.35" - Unchanged
"tf_feign_death_duration": "0.0"
"tf_feign_death_speed_duration": "0.0"
"tf_stealth_damage_reduction": "0.1"

I can't simply change the speed duration to 0.0, because it seems to link the feign death buff to it as well. So if I set the duration to 0.0, the speed buff instantly goes away, and so does the feign death buff, even if it's set to last longer. So I have two options, remove the speed buff through SourceMod, or set damage based on watch index, take your pick.

**Comment by Chdata** [comment]
speedbuff through sourcemod using that snippet i pmed to you the other day

**Comment by Starblaster64** [comment]
@FlaminSarge Since you offered I may as well ask. I don't know how well acquainted you are with methodmaps (hopefully more than me), but do you have any clue as to why @WildCard65's code isn't modifying (some) CVARs?

514ca75 & a0c3a61 are the latest relevant commits.

tf_arena_use_queue and tf_scout_hype_pep_max appear to be the only CVARs that are modified in-game.

**Comment by Chdata** [comment]
tf_scout_hype_pep_max, I don't think that's even used for anything in VSH anymore. That's related to the soda popper isn't it? That was from ages before the soda popper effect changed.

**Comment by Starblaster64** [comment]
Actually it was for the BFB. I think Valve had the Soda Popper meter and BFB meter in the same thing or something, because letting the BFB hit 100% charge made it trigger the Soda Popper effect, which was a really nice way to deal with it actually. 100% minicrits, constantly draining your boost and you couldn't gain/lose extra amounts by damage or jumping.

**Comment by Chdata** [comment]
oh wow that actually worked, it was always nonfunctional for me.

there's an attribute to drain hype by the float you specify which I was using

**Comment by FlaminSarge** [comment]
I always thought the "BFB hits 100% = gain a different bonus and Hype starts draining" was an elegant solution to the whole deal...
I'm not sure why people opted to just *remove* it instead of trying that solution.

**Comment by Chdata** [comment]
I tested it once and it didn't seem to do anything, that's why it got removed. Maybe that gimmick got broken by Valve at some point?

**Comment by Starblaster64** [comment]
"broken by Valve at some point?"
I just said that :v

**Comment by FlaminSarge** [comment]
It's really easy to just check "m_flHypeMeter" >= 99.0, I just can't remember what prop activates the drain on it. Will look sometime tomorrow, I guess.

**Comment by Chdata** [comment]
My change to BFB merely made the hype constantly drain all the time with no extra buffs. I kinda like it, if you have very consistent aim it's good. But you also have to account for reload.

But nobody uses over soda popper which gives crits or back scatter which gives back crits.

**Comment by Starblaster64** [comment]
I believe that's everything for now, unless I've forgotten something.

**Comment by WildCard65** [comment]
2 builds failed to properly get to running spcomp, re-running them now.

File: addons/sourcemod/scripting/saxtonhale.sp [hunk 2]
```diff
@@ -865,19 +871,18 @@ public void OnConfigsExecuted()
         mp_teams_unbalance_limit = cvarMPUnbalanceLimit.IntValue;
         tf_arena_first_blood = cvarTFFirstBlood.IntValue;
         mp_forcecamera = cvarMPForceCamera.IntValue;
-        tf_scout_hype_pep_max = cvarTFScoutHypeMax.FloatValue;
-        cvarTFUseQueue.IntValue = 0;
-        cvarMPUnbalanceLimit.IntValue = TF2_GetRoundWinCount() ? 0 : 1; // s_bLateLoad ? 0 :
+        tf_dropped_weapon_lifetime = cvarTFWeaponLifeTime.IntValue;
+		tf_feign_death_activate_damage_scale = cvarTFFeignActivateDamageScale.FloatValue;
```
Comment (WildCard65) [inline addons/sourcemod/scripting/saxtonhale.sp]: Indentation needs fixing here as well as the 3 lines below this one.

File: addons/sourcemod/scripting/saxtonhale.sp [hunk 3]
```diff
@@ -1546,11 +1546,21 @@ public Action event_round_start(Event event, const char[] name, bool dontBroadca
...
+        cvarTFFeignDamageScale.SetFloat(tf_feign_death_damage_scale);
+        cvarTFFeignDeathDuration.SetInt(tf_feign_death_duration);
+        cvarTFStealthDamageReduction.SetFloat(tf_stealth_damage_reduction);
         CreateTimer(71.0, Timer_EnableCap, _, TIMER_FLAG_NO_MAPCHANGE);
         return Plugin_Continue;
     }
-    FindConVar("mp_teams_unbalance_limit").IntValue =  TF2_GetRoundWinCount() ? 0 : 1; // s_bLateLoad ? 0 :
+    cvarMPUnbalanceLimit.SetInt(TF2_GetRoundWinCount() ? 0 : 1); // s_bLateLoad ? 0 :
+    cvarTFWeaponLifeTime.SetInt(TF2_GetRoundWinCount() ? 0 : tf_dropped_weapon_lifetime);
+	cvarTFFeignActivateDamageScale.SetFloat(TF2_GetRoundWinCount() ? 0.1 : tf_feign_death_activate_damage_scale);
```
Comment (WildCard65) [inline addons/sourcemod/scripting/saxtonhale.sp]: Indentation here as well, same with the next 3 lines.

**Comment by Starblaster64** [comment]
Seems all the builds trying to build with Goomba Stomp failed, I'm pretty sure I didn't touch anything related to that though.

**Comment by WildCard65** [comment]
@Starblaster64 If you ever need a build restarted because of an error where build.sh wasn't able to get to running spcomp, let me know either via Steam or github.

File: addons/sourcemod/scripting/saxtonhale.sp [hunk 4]
```diff
@@ -5007,11 +5069,14 @@ public Action OnTakeDamage(int client, int &attacker, int &inflictor, float &dam
...
             }
-            else if (TF2_IsPlayerInCondition(client, TFCond_Cloaked) && TF2_IsPlayerInCondition(client, TFCond_DeadRingered))
+            else if (TF2_IsPlayerInCondition(client, TFCond_Cloaked))
             {
                 if (damagetype & DMG_CRIT)
                     damagetype &= ~DMG_CRIT;
-                damage = 850.0;
+                if (TF2_IsPlayerInCondition(client, TFCond_DeadRingered))
+                    damage = 620.0;
+                else if (!TF2_IsPlayerInCondition(client, TFCond_DeadRingered))
```
Comment (50Wliu) [inline addons/sourcemod/scripting/saxtonhale.sp]: This can probably just be an `else`, right?

Comment (WildCard65) [inline addons/sourcemod/scripting/saxtonhale.sp]: Yes

**Comment by WildCard65** [comment]
Hey @Starblaster64 mind putting your git name inside "(" and ")" in the changelog messages related to the changes in this pr.

**Comment by FlaminSarge** [comment]
What's the max amount of text allowed in a single panel, again?

**Comment by Chdata** [comment]
panels 512

**Comment by shadow93** [comment]
Might want to correct FileExists to add the second argument (set it to true)

Otherwise, it's basically adding stuff to downloads table if they don't exist.

**Comment by WildCard65** [comment]
@Starblaster64 You should probably change around changelog, there appearing for this pr in reverse order.

File: addons/sourcemod/scripting/saxtonhale.sp [hunk 5]
```diff
@@ -6064,11 +6176,29 @@ void FindVersionData(Panel panel, int versionindex)
...
+        }
+        case 71: //1.53
+        {
+            panel.DrawText("8) Disabled dropping weapons during VSH rounds.(Starblaster64)");
+            panel.DrawText("9) OVERRIDE_MEDIGUNS_ON is now on by default. Mediguns will have their stats overriden instead of being replaced by a custom medigun.(Starblaster64)");
+            panel.DrawText("10) Natascha will no longer keep its bonus ammo when being replaced.(Starblaster64)");
+            panel.DrawText("11) Unnerfed the Easter Bunny's rage.(Chdata)");
+            panel.DrawText("12) Diamondback revenge crits on stab reduced from 3 -> 2.(Chdata)");
+            panel.DrawText("13) Fixed not all Soldier guns minicritting airborne targets.(Starblaster64)");
+            panel.DrawText("14) Updated English translation phrases, class info, etc.(Starblaster64)");
```
Comment (WildCard65) [inline addons/sourcemod/scripting/saxtonhale.sp]: This line renders as "Up" only

**Comment by FlaminSarge** [comment]
Somebody should figure out a way to make all Mediguns use the Kritz taunt. Maybe using a response context.

**Comment by Starblaster64** [comment]
@WildCard65 I have no idea why I didn't catch that sooner.

**Comment by WildCard65** [comment]
@Starblaster64 I'm going to merge this now since @shadow93 says everything is fair.

**Pull request merged**